beacon/beacon.go

```diff
@@ -776,17 +764,19 @@ func (pd *ProtocolDriver) runProtocol(ctx context.Context, epoch types.EpochID,
 
 func calcBeacon(logger log.Log, set proposalSet) types.Beacon {
 	allProposals := set.sort()
-	allHexes := make([]string, len(allProposals))
-	for i, h := range allProposals {
-		allHexes[i] = hex.EncodeToString(h[:])
-	}
+
 	// Beacon should appear to have the same entropy as the initial proposals, hence cropping it
 	// to the same size as the proposal
 	beacon := types.BytesToBeacon(allProposals.hash().Bytes())
 	logger.With().Info("calculated beacon",
 		beacon,
 		log.Int("num_hashes", len(allProposals)),
-		log.String("proposals", strings.Join(allHexes, ", ")),
+		log.Array("proposals", zapcore.ArrayMarshalerFunc(func(enc zapcore.ArrayEncoder) error {
+			for _, h := range allProposals {
+				enc.AppendString(hex.EncodeToString(h[:]))
+			}
+			return nil
+		})),
```

Comment:
Adding a small method to `proposalList` would allow to log `allProposals` directly:

```go
func (hl proposalList) MarshalLogArray(enc zapcore.ArrayEncoder) error {
	for _, proposal := range hl {
		enc.AppendString(hex.EncodeToString(proposal[:]))
	}
	return nil
}
```

```suggestion
		log.Array("proposals", allProposals),
	),
```